engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -117,40 +167,39 @@ struct blessed_hammer_tick_t : public paladin_spell_t
   void impact( action_state_t* s ) override
   {
     paladin_spell_t::impact( s );
-
     // apply BH debuff to target_data structure
-    td( s -> target ) -> debuff.blessed_hammer -> trigger();
+    // Does not interact with vers.
+    // To Do: Investigate refresh behaviour
+    td( s -> target ) -> debuff.blessed_hammer -> trigger(
+      1,
+      s -> attack_power * p() -> talents.blessed_hammer -> effectN( 1 ).percent()
+    );
   }
 };
 
 struct blessed_hammer_t : public paladin_spell_t
 {
   blessed_hammer_tick_t* hammer;
-  int num_strikes;
+  double num_strikes;
 
   blessed_hammer_t( paladin_t* p, const std::string& options_str ) :
     paladin_spell_t( "blessed_hammer", p, p -> talents.blessed_hammer ),
     hammer( new blessed_hammer_tick_t( p ) ), num_strikes( 2 )
   {
-    add_option( opt_int( "strikes", num_strikes ) );
+    add_option( opt_float( "strikes", num_strikes) );
     parse_options( options_str );
 
-    // Sane bounds for num_strikes - only makes three revolutions, impossible to hit one target more than 3 times.
-    // Likewise calling the spell with 0 or negative strikes is sort of pointless.
-    if ( num_strikes < 1 )
+    // Sanity check for num_strikes
+    if ( num_strikes <= 0 || num_strikes > 10)
     {
-      num_strikes = 1;
-      sim -> error( "{} trying to hit less than one time with blessed_hammer, value changed to 1", p -> name() );
-    }
-    if ( num_strikes > 3 )
-    {
-      num_strikes = 3;
-      sim -> error( "{} trying to hit more than three times with blessed_hammer, value changed to 3", p -> name() );
+      num_strikes = 2;
+      sim -> error( "{} invalid blessed_hammer strikes, value changed to 2", p -> name() );
     }
 
-    dot_duration = 0_ms; // The periodic event is handled by ground_aoe_event_t
+    // dot_duration = 0_ms; // The periodic event is handled by ground_aoe_event_t
+
     may_miss = false;
-    base_tick_time = 1667_ms; // Rough estimation based on stopwatch testing
+    base_tick_time = data().duration();
```

Comment:
Putting the duration in base_tick_time and using it to compute the actual pulse time of the ground aoe isn't very transparent, you should probably set both base_tick_time and dot_duration to 0 in the constructor and use data().duration() later in pulse time calculations.